pkg/scaling/executor/scale_jobs.go

```diff
@@ -36,13 +36,15 @@ func (e *scaleExecutor) RequestJobScale(ctx context.Context, scaledJob *kedav1al
 		scaledJob.Status.LastActiveTime = &now
 		e.updateLastActiveTime(ctx, e.logger, scaledJob)
 		e.createJobs(scaledJob, scaleTo, effectiveMaxScale)
-		err := e.cleanUp(scaledJob)
-		if err != nil {
-			e.logger.Error(err, "Failed to cleanUp jobs")
-		}
 	} else {
 		e.logger.V(1).Info("No change in activity")
 	}
+
+	err := e.cleanUp(scaledJob)
```

Comment:
i wonder if cleaning up should have its own timer and not rely on the pollingInterval timer. might make the behavior a bit more predictable,just a thought. we shouldnt block the PR on it

Reply:
@ahmelsayed 
That's make sense. I'll consider to implement it on the next PR.  Should be different loop and interval might be cooler than this. :)